File: src/api/src/Interviewd.Application.Dto/InterviewDto.cs
using System.Collections.Generic;
using Interviewd.Common;

namespace Interviewd.Application.Dto
{
    public class InterviewDto : IIdentifiable
    {
        public string Id { get; set; }

        public CandidateDto Candidate { get; set; }

        public IEnumerable<QuestionDto> Questions { get; set; }
    }
}

File: src/android/app/src/main/java/io/github/alexdenton/interviewd/api/dto/InterviewDto.kt
package io.github.alexdenton.interviewd.api.dto

import io.github.alexdenton.interviewd.entities.Interview
import io.github.alexdenton.interviewd.entities.InterviewStatus

/**
 * Created by ryan on 9/9/17.
 */
data class InterviewDto (
        val id: Int,
        val candidate: CandidateDto,
        val name: String,
        val questions: List<QuestionDto>,
        var status: InterviewStatus
){
    fun toInterview() = Interview(id, candidate.toCandidate(), name, questions.map { it.toQuestion() }, status)
}